### ISSUE TITLE
rel_path not used (Trac #82)

### ISSUE THREAD

**Issue by mnot** [httpwg/httpbis-issues#82]
RFC2616 changed the ABNF for http_URL so that it doesn't use rel_path (as defined in RFC2396) anymore.

However, that definition is still "adopted" in section 3.2.1;
            
  URIs in HTTP can be represented in absolute form or relative to some known base URI [11], depending upon the context of their use. The two forms are differentiated by the fact that absolute URIs always begin with a scheme name followed by a colon. For definitive information on URL syntax and semantics, see "Uniform Resource Identifiers (URI): Generic Syntax and Semantics," RFC 2396 [42] (which replaces RFCs 1738 [4] and RFC 1808 [11]). This specification adopts the definitions of "URI-reference", "absoluteURI", "relativeURI", "port", "host","abs_path", "rel_path", and "authority" from that specification."
        
...and used in section 13.9.p.2:
                
  "We note one exception to this rule: since some applications have traditionally used GETs and HEADs with query URLs (those containing a "?" in the rel_path part) to perform operations with significant side effects, caches MUST NOT treat responses to such URIs as fresh unless the server provides an explicit expiration time. This specifically means that responses from HTTP/1.0 servers for such URIs SHOULD NOT be taken from a cache. See Section 9.1.1 for related information."

Migrated from https://trac.ietf.org/ticket/82
```json
{
    "status": "closed",
    "changetime": "2008-01-24T12:46:28",
    "_ts": "1201178788000000",
    "description": "RFC2616 changed the ABNF for http_URL so that it doesn't use rel_path (as defined in RFC2396) anymore.\n\nHowever, that definition is still \"adopted\" in section 3.2.1;\n            \n  URIs in HTTP can be represented in absolute form or relative to some known base URI [11], depending upon the context of their use. The two forms are differentiated by the fact that absolute URIs always begin with a scheme name followed by a colon. For definitive information on URL syntax and semantics, see \"Uniform Resource Identifiers (URI): Generic Syntax and Semantics,\" RFC 2396 [42] (which replaces RFCs 1738 [4] and RFC 1808 [11]). This specification adopts the definitions of \"URI-reference\", \"absoluteURI\", \"relativeURI\", \"port\", \"host\",\"abs_path\", \"rel_path\", and \"authority\" from that specification.\"\n        \n...and used in section 13.9.p.2:\n                \n  \"We note one exception to this rule: since some applications have traditionally used GETs and HEADs with query URLs (those containing a \"?\" in the rel_path part) to perform operations with significant side effects, caches MUST NOT treat responses to such URIs as fresh unless the server provides an explicit expiration time. This specifically means that responses from HTTP/1.0 servers for such URIs SHOULD NOT be taken from a cache. See Section 9.1.1 for related information.\"\n\n",
    "reporter": "mnot@pobox.com",
    "cc": "",
    "resolution": "fixed",
    "time": "2007-12-22T03:37:51",
    "component": "p1-messaging",
    "summary": "rel_path not used",
    "priority": "",
    "keywords": "",
    "milestone": "02",
    "owner": "julian.reschke@gmx.de",
    "type": "editorial",
    "severity": ""
}
```

**Comment by mnot** [comment]
Trac update at 20071222T03:38:34: mnot@pobox.com commented:

> Proposal:
> 
>  * get rid of the mention in 3.2.1, and
>  * in 13.9 paragraph 2, replace "...query URLs (those containing a "?" in the rel_path part)..." with "...URLs containing a query part..."

**Comment by mnot** [comment]
Trac update at 20080106T18:56:22: julian.reschke@gmx.de commented:

> I think we can resolve that right away as proposed in <http://greenbytes.de/tech/webdav/draft-lafon-rfc2616bis-latest.html#rfc.issue.i82-rel_path-not-used>.

**Comment by mnot** [comment]
Trac update at 20080124T12:37:38: 
* julian.reschke@gmx.de changed owner from "fielding@gbiv.com" to "julian.reschke@gmx.de"
* julian.reschke@gmx.de changed status from "assigned" to "new"

**Comment by mnot** [comment]
Trac update at 20080124T12:46:28: 
* julian.reschke@gmx.de commented:

> Fixed in [173]:
> 
> Resolve #82: do not use rel_path in prose anymore, it hasn't been used inside the ABNF for http_URL since 2616 (closes #82).


* julian.reschke@gmx.de changed resolution from "" to "fixed"
* julian.reschke@gmx.de changed status from "new" to "closed"
